lib/services/local_sync_service.dart

```diff
@@ -49,11 +50,20 @@ class LocalSyncService {
     PhotoManager.startChangeNotify();
   }
 
-  Future<void> sync() async {
+  Future<void> sync({bool isAppInBackground = false}) async {
     if (!_prefs.containsKey(kHasGrantedPermissionsKey)) {
       _logger.info("Skipping local sync since permission has not been granted");
       return;
     }
+
+    if (Platform.isAndroid && !isAppInBackground) {
+      var permissionState = await PhotoManager.requestPermissionExtend();
+      if (permissionState != PermissionState.authorized) {
+        _logger.severe(
+            "sync requested with invalid permission", permissionState);
```

Comment:
Shouldn't we call `permissionState.toString()`? Not sure if it will get serialized otherwise.

Reply:
> Shouldn't we call `permissionState.toString()`? Not sure if it will get serialized otherwise.
Will change it. 
fyi: currently it's getting logged as
```
      I   [LocalSyncService] [SEVERE] [2021-09-05 15:25:57.404439] sync requested with invalid permission
                         I  ⤷ type: PermissionState
                         I  ⤷ error: PermissionState.denied
```